modules/custom/az_media/az_media.module

```diff
@@ -148,3 +149,18 @@ function az_media_theme() {
     ],
   ];
 }
+
+/**
+ * Implements hook_entity_presave().
+ *
+ * This allows content editors to override remote video thumbs.
+ */
+function az_media_entity_presave(EntityInterface $entity) {
+  if (
+    $entity->getEntityTypeId() === 'media'
+    && $entity->bundle() === 'az_remote_video'
+    && isset($entity->field_media_az_image->entity)
+    ) {
```

Comment:
Nitpick:  Line breaks like these in control structures are discouraged in the coding standards:
https://www.drupal.org/docs/develop/standards/coding-standards#linelength

```suggestion
  if ($entity->getEntityTypeId() === 'media' && $entity->bundle() === 'az_remote_video' && isset($entity>field_media_az_image->entity)) {
```